drivers/dome/scopedome_dome.cpp

```diff
@@ -100,103 +109,64 @@ bool ScopeDome::initProperties()
+                            "Home sensor position", SITE_TAB, IP_RW, 60, IPS_OK);
+
+    HomeSensorPolaritySP[ScopeDomeCard::ACTIVE_HIGH].fill("ACTIVE_HIGH", "Active high", ISS_ON);
+    HomeSensorPolaritySP[ScopeDomeCard::ACTIVE_LOW].fill("ACTIVE_LOW", "Active low", ISS_OFF);
+    HomeSensorPolaritySP.fill(getDeviceName(), "HOME_SENSOR_POLARITY",
+                              "Home sensor polarity", SITE_TAB, IP_RW, ISR_1OFMANY, 0, IPS_OK);
+
+    FindHomeSP[0].fill("START", "Start", ISS_OFF);
+    FindHomeSP.fill(getDeviceName(), "FIND_HOME", "Find home", MAIN_CONTROL_TAB, IP_RW,
+                    ISR_ATMOST1, 0, IPS_OK);
+
+    DerotateSP[0].fill("START", "Start", ISS_OFF);
+    DerotateSP.fill(getDeviceName(), "DEROTATE", "Derotate", MAIN_CONTROL_TAB, IP_RW,
+                    ISR_ATMOST1, 0, IPS_OK);
+
+    FirmwareVersionsNP[0].fill("MAIN", "Main part", "%2.2f", 0.0, 99.0, 1.0, 0.0);
+    FirmwareVersionsNP[1].fill("ROTARY", "Rotary part", "%2.1f", 0.0, 99.0, 1.0, 0.0);
+    FirmwareVersionsNP.fill(getDeviceName(), "FIRMWARE_VERSION",
+                            "Firmware versions", INFO_TAB, IP_RO, 60, IPS_IDLE);
+
+    StepsPerRevolutionNP[0].fill("STEPS", "Steps per revolution", "%5.0f", 0.0, 99999.0, 1.0, 0.0);
+    StepsPerRevolutionNP.fill(getDeviceName(), "CALIBRATION_VALUES",
+                              "Calibration values", SITE_TAB, IP_RO, 60, IPS_IDLE);
+
+    CalibrationNeededSP[0].fill("CALIBRATION_NEEDED", "Calibration needed", ISS_OFF);
+    CalibrationNeededSP.fill(getDeviceName(), "CALIBRATION_STATUS",
+                             "Calibration status", SITE_TAB, IP_RO, ISR_ATMOST1, 0, IPS_IDLE);
+
+    StartCalibrationSP[0].fill("START", "Start", ISS_OFF);
+    StartCalibrationSP.fill(getDeviceName(), "RUN_CALIBRATION", "Run calibration",
+                            SITE_TAB, IP_RW, ISR_ATMOST1, 0, IPS_OK);
+
+    CredentialsTP[0].fill("USERNAME", "User name", "scopedome");
+    CredentialsTP[1].fill("PASSWORD", "Password", "default");
+    CredentialsTP.fill(getDeviceName(), "CREDENTIALS", "Credentials", CONNECTION_TAB, IP_RW, 0, IPS_OK);
 
     SetParkDataType(PARK_AZ);
 
     addAuxControls();
 
-    // Set serial parameters
-    serialConnection->setDefaultBaudRate(Connection::Serial::B_115200);
+    CardTypeSP[0].fill("USB_CARD_21", "USB Card 2.1", ISS_ON);
+    CardTypeSP[1].fill("ARDUINO", "Arduino Card", ISS_OFF);
+    CardTypeSP.fill(getDeviceName(), "CARD_TYPE", "Card type",
+                    CONNECTION_TAB, IP_RW, ISR_1OFMANY, 0, IPS_OK);
+
+    defineProperty(&CardTypeSP);
+    if(!loadConfig(true, CardTypeSP.getName()))
+    {
+        // Set serial parameters
+        // This is tricky as Arduino card communicates at 9600 bauds, but
+        // USB Card 2.1 uses 115200 bauds
+        serialConnection->setDefaultBaudRate(Connection::Serial::B_115200);
+    }
+
+    defineProperty(CredentialsTP);
+    loadConfig(true, CredentialsTP.getName());
```

Comment:
Maybe IUGetConfigText ?

Reply:
Same comment here, could be cleaner without repeating the names, opportunity for future usability improvement